### PR TITLE
Standup reports

### PR DESCRIPTION
**Pull request by bpostlethwaite**
Almost complete the persistent standup reporting. Notes of what still requires implementation following. Might get to that today.

@scjody @BRONSOLO 

@abourget will be hard forking plotbot to remake it with Slack and also overhauling plugin architecture (both things on my dream list had I time - so this is great news). I am hoping that we will be able to later use the new bot, fork off the new bot or at very least re-hard fork it and build off of that with due credit. 

( @doeg for your interest )

### CODE REVIEW

**Comment by bpostlethwaite** [comment]
Okay @scjody and @BRONSOLO for your review.

**Comment by bpostlethwaite** [comment]
Closing this branch to clean up the vendoring diffs

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,32 @@
...
+# Folders
+_obj
+_test
+
+# Architecture specific extensions/prefixes
+*.[568vq]
+[568vq].out
+
+*.cgo1.go
+*.cgo2.c
```
Comment (abourget) [inline .gitignore]: why those ignores? does `goleveldb` require cgo ?

Comment (bpostlethwaite) [inline .gitignore]: Nah I just copied the .gitignore from githubs auto ignore file repo. I only needed the file.test addition but decided we may as well stick in the whole kitchen sink.
https://github.com/github/gitignore

Comment (bpostlethwaite) [inline .gitignore]: Also I am pretty sure goleveldb is a pure go implementation of leveldb.

File: standup/standupData.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+		for _, user := range users {
+			seenUsers[user.Email] = user
+		}
+	}
+
+	// write header depending on single or multiple user case
+	if len(seenUsers) == 1 {
+		singleUserReport = true
+		// only one user in the map, write the header for this user
+		for _, user := range seenUsers {
```
Comment (scjody) [inline standup/standupData.go]: :cow2: Kinda weird to use a for loop here, isn't it?

**Comment by scjody** [comment]
:dancer: 

Also yay Slack!  It's a lot better in many little ways that all add up!

File: standup/standupData.go (hunk 2 above)
Comment (bpostlethwaite) [inline standup/standupData.go]: oh yeh, that was hard to write - I winced. I figured I would be refactoring all that, turns out I didn't have to. I'll fix it up with a little state variable now

**Pull request merged**